tests/rules/taint_lambda_cpp.cpp

```diff
@@ -28,12 +28,10 @@ void correct_in_lambda() {
     };
 }
 
-void todo_only_sink_in_lambda(int *p) {
-    // We assume that an unused lambda is called at the end of the parent
-    // function.
+void only_sink_in_lambda(int *p) {
     source(p);
     auto f2 = [&p]() {
-        // ruleid: source-sink
+        // ok: source-sink
         sink(p);
```

Comment:
Should this be an `ok` ? Is this is related to the regression ?

Reply:
I adjusted the logic slightly while addressing the regression. Now, when we place unused lambdas at the end of the parent function, they no longer have in-edges. This change also resolves the todo item in the test.

Reply:
TBH when I asked to add this test I actually wanted it to be a `ruleid`, we have a lambda that when evaluated it's very likely to cause taint to go into a sink. I could be wrong but I think this deserves a separate discussion. Could we separate fixing the regression from changing the behavior here?

Reply:
done.